src/MetadataManager.actor.cpp

```diff
@@ -121,7 +121,7 @@ constructContext(Namespace ns, Reference<DocTransaction> tr, DocumentLayer* docL
 
 			for (const auto& indexObj : allIndexes) {
 				IndexInfo index = MetadataManager::indexInfoFromObj(indexObj, cx);
-				if (index.status <= IndexInfo::IndexStatus::BUILDING) {
+				if (index.status != IndexInfo::IndexStatus::INVALID) {
```

Comment:
IMHO, I think the original one is better. In case we added anything between `BUILDING` and `INVALID` and we do not want to include indexes in that new state, this line becomes a bug and would be really hard to find out.

Reply:
I think if we add any new state we have to revisit this code anyway. As it is code looks confusing, I am happy to change it back if you disagree.

Reply:
I am fine with changing it.